src/test/java/io/github/eocqrs/qce/User.java

```diff
@@ -0,0 +1,25 @@
+package io.github.eocqrs.qce;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public record User(String name) implements DomainObject<User> {
+
+    private static final Map<String, User> STATES = new HashMap<>(0);
```

Comment:
@l3r8yJ, can you explain, please why we need the STATES?

Reply:
@h1alexbel it's like a local change history, not sure how necessary it is

Reply:
@l3r8yJ maybe we can name it `changelog`? and wrap up the Map<K, T> to Changelog object

WDYT?

Reply:
@h1alexbel yes, good suggestion, but do you think we really need it, overall?

Reply:
@l3r8yJ I think yes, simply because event sourcing can easily enable historical change log of the entire app

Reply:
@h1alexbel okay, I'll try to do something about it

Reply:
Nice